sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -296,6 +296,25 @@ class Column(protected[sql] val expr: Expression) extends Logging {
   def eqNullSafe(other: Any): Column = this <=> other
 
   /**
+   * True if the current column is between the lower bound and upper bound, inclusive.
+   *
+   * @group java_expr_ops
+   */
+  def between(lowerBound: String, upperBound: String): Column = {
+    between(Column(lowerBound), Column(upperBound))
+  }
+
+  /**
+   * True if the current column is between the lower bound and upper bound, inclusive.
+   *
+   * @group java_expr_ops
+   */
+  def between(lowerBound: Column, upperBound: Column): Column = {
+    And(GreaterThanOrEqual(this.expr, lowerBound.expr),
```

Comment:
actually - how about we change the implementation here to use Column directly, i.e.
```scala
(this >= lowerBound) && (this <= upperBound)
```

and the Python one simply calls the JVM function?